lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -141,19 +146,74 @@ def purge
           self
         end
 
-        def read
+        def read(**kwargs)
+          raise ArgumentError, '`compressed: gzip` can be used for Compressable module' if kwargs[:compressed] == :gzip
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def open(&block)
+        def open(**kwargs, &block)
+          raise ArgumentError, '`compressed: gzip` can be used for Compressable module' if kwargs[:compressed] == :gzip
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def write_to(io)
+        def write_to(io, **kwargs)
+          raise ArgumentError, '`compressed: gzip` can be used for Compressable module' if kwargs[:compressed] == :gzip
           open do |i|
             IO.copy_stream(i, io)
           end
         end
+
+        module Decompressable
+          include Fluent::Plugin::Compressable
+
+          def bytesize(**kwargs)
```

Comment:
Do you mean that `chunk.bytesize` should return the bytesize of decompressed data size? It's different from actual bytesize.
With this definition, buffer chunks doesn't grow up to configured byte size (by `chunk_limit_size`).